app/controllers/api/v2/courses_controller.rb

```diff
@@ -176,8 +182,10 @@ def update_subjects
         if request_has_duplicate_subject_ids?
           @course.errors.add(:subjects, :duplicate)
         else
-          @course.subjects = []
+          @previous_subject_names = subject_names_for(@course)
```

Comment:
Also worth noting that the `update_subjects` method is being called via the `create` action causing further complications / nightmares